tests/desktop/developer_hub/test_submit_fxos_add_on.py

```diff
@@ -0,0 +1,65 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import shutil
+import uuid
+from zipfile import ZipFile
+
+import pytest
+
+from pages.desktop.developer_hub.content_tools import ContentTools
+from tests.base_test import BaseTest
+
+
+class TestSubmitFxOSAddOn(BaseTest):
+
+    @pytest.fixture
+    def fxos_add_on_name(self, request, mozwebqa):
+
+        add_on_name = str(uuid.uuid4())
+
+        # backup the file we're changing, just in case!
+        shutil.copyfile(self._get_resource_path('fxaddon/manifest.json'), self._get_resource_path('manifest.bak'))
+
+        # edit the json file
+        with open(self._get_resource_path('fxaddon/manifest.json'), 'r+') as json_file:
+            data = json.load(json_file)
+            data['name'] = add_on_name
+            json_file.seek(0)  # rewind to beginning of file
+            json_file.write(json.dumps(data))  # write the updated version
+            json_file.truncate()  # truncate the remainder of the data in the file
+
+        # update the zip file
+        with ZipFile(self._get_resource_path('fxosaddon.zip'), 'w') as zipfile:
+            for root, dirs, files in os.walk(self._get_resource_path('fxaddon')):
+                for file in files:
+                    zipfile.write(os.path.join(root, file), file)
+
+        def fin():
+            # copy the backup back
+            shutil.copyfile(self._get_resource_path('manifest.bak'), self._get_resource_path('fxaddon/manifest.json'))
+        request.addfinalizer(fin)
+
+        return add_on_name
+
+    @pytest.mark.credentials
+    def test_add_on_submission(self, mozwebqa, new_user, fxos_add_on_name):
+
+        try:
+            page = ContentTools(mozwebqa).go_to_page()
+            page.login(mozwebqa, new_user['email'], new_user['password'])
+            page.click_submit_new_add_on()
+            page.click_agree()
+            page.select_add_on_file(self._get_resource_path('fxosaddon.zip'))
+            page.click_submit_add_on_form_button()
+            assert 'success' in page.notification_message
+            add_on = page.add_on(fxos_add_on_name)
+            assert 'pending' == add_on.status.lower()
+        finally:
```

Comment:
I'm really not keen on the try/finally. As we're using unique users, we're not going to be leaving addons behind for other tests, so is cleanup as important as if we were reusing the same user? I'd like to suggest either deleting the add-on only when the test passes, or attempting the delete as part of a fixture.

The try/finally also messes with our reporting, as any screenshots/html/etc will be after the finally and not show the point of the actual failure.